.github/workflows/coverage.yml

```
@@ -0,0 +1,69 @@
+name: C++ and Python coverage
+
+on: 
+  schedule:
+    - cron: '0 0 * * 0'
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash -l {0}  # required for conda env
+
+jobs:
+  coverage:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install dependencies
+        run: |
+          # We use g++ from system instead of from conda, since we need to ensure that
+          # a matching gcov is used. Otherwise coverage results are incomplete/broken.
+          sudo apt install --yes lcov g++-11
+          sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 100
+          sudo update-alternatives --install /usr/bin/gcov gcov /usr/bin/gcov-11 100
+          sudo update-alternatives --set g++ "/usr/bin/g++-11"
+          sudo update-alternatives --set gcov "/usr/bin/gcov-11"
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+          fetch-depth: 0  # history required so cmake can determine version
+
+      - name: Disable TBB for much faster compilation.
+        run: cat .buildconfig/ci-linux.yml | grep -v 'gxx\|tbb' > env.yml
+
+      - name: Setup conda environment
+        uses: mamba-org/provision-with-micromamba@main
+        with:
+          micromamba-version: 0.23.0
```

```suggestion
          micromamba-version: 0.24.0
```
Or even 25 which was released yesterday. Didn't test it yet, but used 24 in scippneutron in the past.

---

> What about other packages? Would we have separate coverage repos for scippnexus, etc? Or would it be better to keep them together but on separate branches. If we keep them separate, I see no reason not to use main.

We cannot use separate branches, since we cannot deploy the GH pages then. We could use subdirs, but I would want to avoid worrying about race conditions and other potential problems.

---

Ok. Wrong thread, though :P

---

I will change to deploy to `main` then.